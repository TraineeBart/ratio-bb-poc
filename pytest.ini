[pytest]
pythonpath = src
testpaths = tests
addopts = --maxfail=1 --disable-warnings -q
markers =
    snapshot: Syrupy snapshot support