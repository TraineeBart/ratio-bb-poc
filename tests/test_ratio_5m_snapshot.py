# ╭──────────────────────────────────────────────────────────────╮
# │ File: tests/test_ratio_5m_snapshot.py                        │
# │ Module: Snapshot Testing                                     │
# │ Doel: Snapshot-validatie van verrijkte ratio 5m dataset      │
# │ Auteur: Quality EngineerGPT                                  │
# │ Laatste wijziging: 2025-07-01                                │
# │ Status: stable                                               │
# ╰──────────────────────────────────────────────────────────────╯

import pandas as pd
from pathlib import Path
import pytest

@pytest.mark.skip(reason="File only available on VPS, not in CI environment")
def test_ratio_5m_column_headers_snapshot(snapshot):
    """
    🧠 Functie: test_ratio_5m_column_headers_snapshot
    Verifieert of de kolomstructuur van de verrijkte ratio 5m CSV overeenkomt met de opgeslagen snapshot.

    ▶️ In:
        - snapshot: Syrupy snapshot fixture

    ⏺ Out:
        - Assertion tegen bekende kolomlijst
    """
    file_path = Path("/opt/ratio-bb-poc/data/signals_bb_ratio_5m.csv")
    df = pd.read_csv(file_path)
    actual_columns = list(df.columns)

    assert actual_columns == snapshot