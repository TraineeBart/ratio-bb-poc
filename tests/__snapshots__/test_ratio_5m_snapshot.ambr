# serializer version: 1
# name: test_ratio_5m_column_headers_snapshot
  list([
    'timestamp',
    'close',
    'rsi',
    'sma_rsi',
    'sma_9',
    'sma',
    'upper_band',
    'lower_band',
    'ratio_lower',
    'ratio_upper',
    'signal',
  ])
# ---
